### ISSUE TITLE
Add `Data.List.First`

### ISSUE THREAD

**Issue by gallais** [agda/agda-stdlib#482]
May be worth refactoring `Any` as an instance of `First`.

```agda
data First (P : Pred A p) (Q : Pred A q) : Pred (List A) (p \lub q) where
  here  : P x -> First P Q (x :: xs)
  there : Q x -> First P Q xs -> First P Q (x :: xs)
```

Typical use case: `Decidable P -> Decidable (First P (¬ \o P))`.

**Comment by HuStmpHrrr** [comment]
How do you want to express `Any` in terms of `First`?

**Comment by gallais** [comment]
If you pick `P` to be constantly the unit type then `First P Q = Any Q`.

But now that I have actually implemented the thing (cf. #483), I'm not sure it's worth
going through the refactoring TBH.

**Comment by HuStmpHrrr** [comment]
I guess you mean `Q` to be the unit type? 

It would be a pain to refactor though, considering `Any` is probably pattern matched a lot and `First` got a higher arity, so I would say it might not worth the effort.

probably an equivalence relation can be proved instead?

**Comment by gallais** [comment]
> I guess you mean `Q` to be the unit type?

You're right. In the actual implementation I have reverse roles for P & Q hence the confusion.
